string_manipulation.rb

```diff
@@ -1,21 +1,28 @@
 # A method to reverse a string in place.
 
 def string_reverse(my_string)
-  i = 0
-  j = my_string.length - 1
+  i = 0 # O(1) constant since variable is set to 0
+  j = my_string.length - 1 # 0(1) constant since j is set to the length of string - 1 (will always be the case regardless since it's fixed even if string is 1000 characters or 10, it will still be length -1)
 
-  while i < j
-    temp = my_string[i]
+  while i < j # while loop happens n amount of times depending on the length of my_string
+    temp = my_string[i] # O(1)
 
-    my_string[i] = my_string[j]
-    my_string[j] = temp
- 
-    i += 1
-    j -= 1
+    my_string[i] = my_string[j] # O(1)
+    my_string[j] = temp # O(1)
+
+    i += 1  # n times (operation: i + 1)
+    j -= 1  # n times (operation: j + 1)
   end
   return my_string
```

Comment:
Since you're reversing the string in place, you don't need to return the string.